app/controllers/homebuyings_controller.rb

```diff
@@ -42,40 +42,49 @@ def create
   end
 
   def show
-    @homebuying = Homebuying.find(params[:id])
+    @homebuying = Homebuying.find(params[:id]) if current_client
   end
 
   def edit
-    if current_client
-      @homebuying = current_client.homebuying
-    elsif current_user
-      @foreclosure = Homebuying.find(params[:id])
-    end
+    @homebuying = current_client.homebuying if current_client
+    # if current_client
+    #   @homebuying = current_client.homebuying
+    #
+    # else
+    #   @foreclosure = Homebuying.find(params[:id])
+    # end
   end
 
   def update
     @homebuying = Homebuying.find(params[:id]) if current_user
     @homebuying = current_client.homebuying if current_client
 
+    @homebuying.update(homebuying_params)
+
     if @homebuying.update(homebuying_params)
-      flash[:success] = "homebuying application submitted."
-      redirect_to "/clients/#{@homebuying.client_id}"
+      flash[:success] = "homebuying application updated"
+      redirect_to "/homebuyings/#{@homebuying.id}"
```

Comment:
Can you update the `homebuying` controller test: `it "redirects to that clients status page" do` with the new redirect?